src/System.Net.Http/src/System/Net/Http/Unix/CurlCallbacks.cs

```diff
@@ -414,5 +428,76 @@ private static void CheckForCompletedTransfers(SafeCurlMultiHandle multiHandle,
                 }
             }
         }
+
+        private async static void PollFunction(SafeCurlMultiHandle multiHandle)
+        {
+            while (true)
+            {
+                if (multiHandle.CancellationToken.IsCancellationRequested)
+                {
+                    return;
+                }
+
+                var count = multiHandle.FdList.Count;
+                if (count == 0)
+                {
+                    await Task.Delay(100).ConfigureAwait(false);
+                    continue;
+                }
+
+                int result = -1;
+                unsafe
+                {
+                    Interop.libc.pollfd* fds = stackalloc Interop.libc.pollfd[count];
+                    lock (multiHandle)
+                    {
+                        if (count != multiHandle.FdList.Count)
+                        {
+                            continue;
+                        }
+                        for (int i = 0; i < count; i++)
+                        {
+                            fds[i].fd = multiHandle.FdList[i];
+                            fds[i].events = PollFlags.POLLIN | PollFlags.POLLOUT;
+                        }
+                    }
+
+                    var retVal = Interop.libc.poll(fds, (uint)count, -1);
```

Comment:
A -1 timeout means it'll block here waiting forever.  Is that correct?  It's going to be blocking a thread pool thread that whole time.